File: sdk/node/src/signals/definitions.js
/**
 * Signal Type Definitions
 * Matches the binary spec in API_SPECIFICATION.md
 */

const SignalType = {
    // Control Flow
    BRANCH: 1,      // 0x01
    SWITCH: 2,      // 0x02
    EXCEPTION: 3,   // 0x03

    // Determinism
    TIME: 4,        // 0x04
    RANDOM: 5,      // 0x05

    // I/O & Context
    IO_READ: 6,     // 0x06
    CTX_START: 7,   // 0x07
    CTX_END: 8,     // 0x08
    DECISION: 9     // 0x09 - Manual Capture
};

module.exports = { SignalType };

File: sdk/go/pkg/signals/definitions.go
package signals

// SignalType defines the type of event captured
type SignalType int

const (
	BRANCH       SignalType = 1
	SWITCH       SignalType = 2
	EXCEPTION    SignalType = 3
	TIME         SignalType = 4
	RANDOM       SignalType = 5
	IO_READ      SignalType = 6
	CTX_START    SignalType = 7
	CTX_END      SignalType = 8
	DECISION     SignalType = 9
)

// Signal represents a single captured event
type Signal struct {
	Type  SignalType  `json:"type"`
	Val   interface{} `json:"val,omitempty"`
	Label string      `json:"label,omitempty"`
	Ctx   string      `json:"ctx,omitempty"`
	TS    int64       `json:"ts"`
}
